core/src/main/scala/org/apache/spark/rdd/PairRDDFunctions.scala

```diff
@@ -70,12 +71,13 @@ class PairRDDFunctions[K, V](self: RDD[(K, V)])
    * In addition, users can control the partitioning of the output RDD, and whether to perform
    * map-side aggregation (if a mapper can produce multiple items with the same key).
    */
-  def combineByKey[C](createCombiner: V => C,
+  def combineByKeyWithClassTag[C](
```

Comment:
Because `PairRDDFunctions` is a stable API, we can't change the method signature of `combineByKey`. Adding the `ClassTag`, would add an implicit argument. If we leave the old `combineByKey` methods and add new `combineByKey` methods with `ClassTag`s, then we get compiler errors being unable to resolve the `combineByKey` symbol.

If you know as way of doing this more cleanly, I would be happy to make that change.